docs/operations/upgrade-gardener.md

```diff
@@ -0,0 +1,87 @@
+# Gardener Upgrade Guide
+
+Upgrading Gardener is a multi-step process that updates its core components, including the operator, control plane, `gardenlet`s, and the clusters it manages (seeds and shoots). To ensure a smooth and safe upgrade, you should follow a specific sequence of steps. This process works with Gardener's built-in reconciliation and rollout mechanisms, which automatically apply changes across your landscape.
+
```

Comment:
I'm not sure where it would be fit best in the structure, but how about adding these point here?

- Operators should apply updates sequentially.
- Operators should ensure that all `Shoot`s reconciled (even if it failed) with the currently installed Gardener version before updating to the next version